src/chttpd/src/chttpd_db.erl

```diff
@@ -222,9 +222,13 @@ changes_callback(waiting_for_updates, Acc) ->
         mochi = Resp1,
         chunks_sent = ChunksSent + 1
     }};
-changes_callback({timeout, _ResponseType}, Acc) ->
+changes_callback({timeout, ResponseType}, Acc) ->
     #cacc{mochi = Resp, chunks_sent = ChunksSent} = Acc,
-    {ok, Resp1} = chttpd:send_delayed_chunk(Resp, "\n"),
+    Chunk = case ResponseType of
+        "eventsource" -> "event: heartbeat\ndata: \n\n";
```

Comment:
Why is this required here? Previously, this seems to be managed at line [171](https://github.com/apache/couchdb/blob/db6a7822b9c5bc501bb20acb6739a1c0cf7f929c/src/chttpd/src/chttpd_db.erl#L171). We are addressing the root issue here or we are duplicating some logic?

Reply:
The thing is, the server was sending plain `\n`s as heartbeat (which is not what is expected for eventsource). You can try it yourself with and without the change. A `GET` `/<db>/_changes?feed=eventsource&heartbeat=1000` should help reproducing the issue.
We may want to remove the lines you've mentioned above, but I don't know the codebase well enough to tell if that would break anything.
I'm hoping I'm not duplicating logic here, but I'm not 100% the previous related code isn't dead code.
IIRC https://github.com/apache/couchdb/pull/3404/files#diff-0ad4fb5d609221dac3b3f5faa48e061bd84f0745ecb0b6953b820d117b55cf5cR106 was the test case that was broken for this.

Reply:
Sorry for the delay. I've been checking this and I think we are leaving dead code during the move to fdb. I suppose this should be addressed in some moment. 

I did some checks changing the callback params to match with the old code [here](https://github.com/apache/couchdb/blob/82bd4d450711b0ff62616229851e5d19fd20e42f/src/chttpd/src/chttpd_changes.erl#L392) and it seems to work with some side effects in the auth cache changes monitor that can be easily solved. 

The old timeout callbacks seem to be dead code, but with the test suite partially I'm not sure. 

Your patch solves the issue in the new execution path, so we will go ahead.